### PR TITLE
Docs/remove docrep

### LINKED ISSUE #490: Remove `docrep` dependency

### PR DESCRIPTION
**Pull request by michalk8**
closes #490

### CODE REVIEW

File: src/moscot/backends/ott/output.py [hunk 1]
```diff
@@ -5,73 +5,72 @@
...
         return_fig: bool = False,
+        ax: Optional[mpl.axes.Axes] = None,
         figsize: Optional[Tuple[float, float]] = None,
         dpi: Optional[int] = None,
         save: Optional[str] = None,
-        ax: Optional[mpl.axes.Axes] = None,
         **kwargs: Any,
     ) -> Optional[mpl.figure.Figure]:
-        """Plot regularized OT costs during the iterations.
+        """Plot regularized :term:`OT` costs during the iterations.
```
Comment (zoepiran) [inline src/moscot/backends/ott/output.py:50]: along the iterations?

File: src/moscot/backends/ott/output.py [hunk 2]
```diff
@@ -100,11 +98,29 @@ def plot_errors(
 
         Parameters
         ----------
-        %(plot_costs.parameters)s
+        last
+            Whether to plot only the ``last`` errors. If :obj:`None`, plot all errors.
```
Comment (zoepiran) [inline src/moscot/backends/ott/output.py:102]: shouldn't be this be like the above [line 55]?
"last
            How many of the last steps of the algorithm to plot. If :obj:`None`, plot the full curve."

File: src/moscot/base/output.py [hunk 3]
```diff
@@ -180,42 +177,37 @@ def sparsify(
...
         value
-            Value to use for sparsification depending on the ``mode``:
-
-                - `'threshold'` - `value` sets the threshold below which entries are set to 0.
-                - `'percentile'` - `value` is the percentile below which entries are set to 0.
-                - `'min_row'` - `value` is not used.
+            Value to use for sparsification.
         batch_size
-            How many rows of the transport matrix to sparsify per batch.
+            How many rows of to materialize when sparsifying the :attr:`transport_matrix`.
```
Comment (zoepiran) [inline src/moscot/base/output.py:200]: redundant "of"

File: src/moscot/problems/generic/_mixins.py [hunk 4]
```diff
@@ -36,46 +34,69 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
...
-        forward: bool = False,  # return value will be row-stochastic if forward=True, else column-stochastic
         aggregation_mode: Literal["annotation", "cell"] = "annotation",
+        forward: bool = False,
         batch_size: Optional[int] = None,
         normalize: bool = True,
         key_added: Optional[str] = _constants.CELL_TRANSITION,
     ) -> pd.DataFrame:
-        """
-        Compute a grouped cell transition matrix.
+        """Compute an aggregate cell transition matrix.
```
Comment (zoepiran) [inline src/moscot/problems/generic/_mixins.py:49]: aggregated?

File: docs/references.rst [hunk 5]
```diff
@@ -1,5 +1,65 @@
 References
 ==========
 
+Bibliography
+------------
 .. bibliography::
     :cited:
+
+Glossary
```
Comment (MUCDK) [inline docs/references.rst:9]: Hi @michalk8 do you want me to fill this in this PR or in a follow-up PR?

File: src/moscot/_types.py [hunk 6]
```diff
@@ -51,8 +51,9 @@
     "sequential",
     "star",
     "external_star",
+    "pairwise",
```
Comment (MUCDK) [inline src/moscot/_types.py:54]: `pairwise` was removed in the constants , `src/moscot/_constants.py`, right? why?

File: src/moscot/base/problems/_mixins.py [hunk 7]
```diff
@@ -458,80 +457,81 @@ def compute_feature_correlation(
...
-        Compute correlation of push or pull distribution with features.
+        """Compute correlation of push-forward or pull-back distribution with features.
+
+        Correlates a feature, e.g., counts of a gene, with probabilities of cells mapped to a set of cells such as
+        the push-forward or pull-back distributions.
 
-        Correlates a feature (e.g. counts of a gene) with probabilities of cells mapped to a set of cells, e.g.
-        a pull back or push forward distribution.
+        .. seealso::
+            - TODO: create and link an example
```
Comment (MUCDK) [inline src/moscot/base/problems/_mixins.py:466]: @ArinaDanilina please open an issue for this and assign yourself

File: src/moscot/base/problems/_mixins.py [hunk 8]
```diff
@@ -458,80 +457,81 @@ def compute_feature_correlation(
...
         corr_method
-            Which type of correlation to compute, options are `pearson`, and `spearman`.
+            Which type of correlation to compute, either ``'pearson'`` or ``'spearman'``.
         significance_method
             Mode to use when calculating p-values and confidence intervals. Valid options are:
 
-                - `fischer` - use Fischer transformation :cite:`fischer:21`.
-                - `perm_test` - use permutation test.
-
+            - ``'fisher'`` - Fischer transformation :cite:`fisher:21`.
```
Comment (MUCDK) [inline src/moscot/base/problems/_mixins.py:477]: typo: `Fischer transformation` -> `Fisher transformation`

File: src/moscot/_types.py (hunk 6 above)
Comment (michalk8) [reply src/moscot/_types.py:54]: Because pairwise policy doesn't exist anymore; this should also be removed here.

File: docs/references.rst (hunk 5 above)
Comment (michalk8) [reply docs/references.rst:9]: Would be great to add it here!

File: src/moscot/backends/ott/_utils.py [hunk 9]
```diff
@@ -49,12 +58,26 @@ def check_shapes(geom_x: Geometry, geom_y: Geometry, geom_xy: Geometry) -> None:
 
 
 def alpha_to_fused_penalty(alpha: float) -> float:
+    """Convert."""
```
Comment (giovp) [inline src/moscot/backends/ott/_utils.py:61]:
```suggestion
    """Convert alpha value as defined in `moscot` to alpha penalty as defined in `ott`."""
```

File: src/moscot/backends/ott/solver.py [hunk 10]
```diff
@@ -200,27 +202,29 @@ def problem_kind(self) -> ProblemKind_t:  # noqa: D102
...
 class GWSolver(OTTJaxSolver):
-    """Solver solving quadratic optimal transport problem.
+    """Solver for the :term:`quadratic problem` :cite:`memoli:2011`.
 
-    The Gromov-Wasserstein (GW) problem involves two distribution in possibly two different spaces.
-    Points in the source distribution are matched to points in the target distribution by comparing the relative
-    location of the points within each distribution.
+    The :term:`Gromov-Wasserstein (GW) <Gromov-Wasserstein>` problem involves two distribution in
+    possibly two different spaces. Points in the source distribution are matched to points in the target distribution
+    by comparing the relative location of the points within each distribution.
```
Comment (giovp) [inline src/moscot/backends/ott/solver.py:215]:
```suggestion
    by comparing the location of the points with respect to their distributions.
```

File: src/moscot/base/output.py [hunk 11]
```diff
@@ -138,14 +135,14 @@ def as_linear_operator(self, scale_by_marginals: bool = False) -> LinearOperator
...
-            Whether to scale by marginals.
+            Whether to scale by :term:`marginals`.
 
         Returns
         -------
         The :attr:`transport_matrix` as a linear operator.
         """
-        push = partial(self.push, scale_by_marginals=scale_by_marginals)
-        pull = partial(self.pull, scale_by_marginals=scale_by_marginals)
+        push = functools.partial(self.push, scale_by_marginals=scale_by_marginals)
```
Comment (giovp) [inline src/moscot/base/output.py:144]: why is this? I notice everywhere that you don't import the top-level object but always refer to the module. This changed from before, could you elaborate on this practice? As in, I understand why in other places but not here.

**Review by giovp — APPROVED** [review]

File: src/moscot/base/problems/birth_death.py [hunk 12]
```diff
@@ -159,46 +148,77 @@ def apoptosis_key(self: BirthDeathProtocol, key: Optional[str]) -> None:
...
+        kwargs
+            Keyword arguments for :func:`~moscot.base.problems.birth_death.beta` and
+            :func:`~moscot.base.problems.birth_death.delta`.
+
+        Returns
+        -------
+        The estimated source or target marginals of shape ``[n,]`` or ``[m,]``, depending on the ``source``.
+        If ``source = True``, also updates the following fields:
+
+        - :attr:`prior_growth_rates` - prior estimate of the source growth rates.
```
Comment (MUCDK) [inline src/moscot/base/problems/birth_death.py:196]: @ArinaDanilina please open an issue to add a link to the corresponding example here.

File: src/moscot/base/problems/birth_death.py [hunk 13]
```diff
@@ -159,46 +148,77 @@ def apoptosis_key(self: BirthDeathProtocol, key: Optional[str]) -> None:
...
         proliferation_key: Optional[str] = None,
         apoptosis_key: Optional[str] = None,
-        marginal_kwargs: Mapping[str, Any] = MappingProxyType({}),
-        **_: Any,
+        **kwargs: Any,
     ) -> ArrayLike:
-        """TODO."""
+        """Estimate the source or target :term:`marginals` with the
+        `birth-death process <https://en.wikipedia.org/wiki/Birth%E2%80%93death_process>`_,
+        as suggested in :cite:`schiebinger:19`.
```
Comment (MUCDK) [inline src/moscot/base/problems/birth_death.py:173]: it does not only implement this method, but also a simpler one. I will adapt the docstrings.

File: src/moscot/base/problems/compound_problem.py [hunk 14]
```diff
@@ -237,87 +278,87 @@ def prepare(
...
         stage: Union[ProblemStage_t, Tuple[ProblemStage_t, ...]] = ("prepared", "solved"),
         **kwargs: Any,
-    ) -> "BaseCompoundProblem[K,B]":
-        """
-        Solve the biological problem.
+    ) -> "BaseCompoundProblem[K, B]":
+        """Solve the individual :term:`OT` subproblems.
+
+        .. seealso:
+            - TODO(michalk8): link an example/tutorial
```
Comment (MUCDK) [inline src/moscot/base/problems/compound_problem.py:296]: @ArinaDanilina please open an issue about this

File: src/moscot/utils/subset_policy.py [hunk 15]
```diff
@@ -51,7 +51,21 @@ def _format(self, value: Any, *, is_source: bool) -> Any:
...
+    adata
+        Annotated data object or a categorical data.
+    key
+        Key in :attr:`~anndata.AnnData.obs` where the categorical data is stored.
+    verify_integrity
+        Whether to check that the data has :math:`\ge 2` categories.
+
+    Examples
+    --------
+    TODO(michalk8): add usage example
```
Comment (MUCDK) [inline src/moscot/utils/subset_policy.py:67]: @ArinaDanilina please open issue and self-assign

File: src/moscot/utils/subset_policy.py [hunk 16]
```diff
@@ -377,18 +512,20 @@ def create_policy(
...
+        Additional keyword arguments.
 
     Returns
     -------
     The policy.
 
     Notes
     -----
-    TODO: link policy example.
+    TODO(michalk8): link policy example.
```
Comment (MUCDK) [inline src/moscot/utils/subset_policy.py:528]: @ArinaDanilina new issue for this please

File: src/moscot/base/problems/compound_problem.py [hunk 17]
```diff
@@ -428,117 +503,57 @@ def add_problem(
...
+
+        - :attr:`problems`
         """
         if TYPE_CHECKING:
             assert isinstance(self._problem_manager, ProblemManager)
         self._problem_manager.remove_problem(key)
         return self
 
-    # TODO(MUCKD): should be on the OT problem level as well
-    def save(
```
Comment (MUCDK) [inline src/moscot/base/problems/compound_problem.py:471]: @michalk8 why do we remove this?

Comment (michalk8) [reply src/moscot/base/problems/compound_problem.py:471]: This is now on the `BaseProblem` level.

Comment (michalk8) [reply src/moscot/base/problems/compound_problem.py:471]: So we that we can also save the `OTProblem`.

File: src/moscot/backends/ott/solver.py (hunk 10 above)
Comment (MUCDK) [reply src/moscot/backends/ott/solver.py:215]: No. this is mathematically wrong. Happy to keep it as it is.

File: src/moscot/base/problems/_mixins.py (hunk 7 above)
Comment (ArinaDanilina) [reply src/moscot/base/problems/_mixins.py:466]: #558

File: src/moscot/utils/subset_policy.py (hunk 16 above)
Comment (ArinaDanilina) [reply src/moscot/utils/subset_policy.py:528]: #561

File: src/moscot/utils/subset_policy.py (hunk 15 above)
Comment (ArinaDanilina) [reply src/moscot/utils/subset_policy.py:67]: #562

File: src/moscot/base/problems/birth_death.py (hunk 12 above)
Comment (ArinaDanilina) [reply src/moscot/base/problems/birth_death.py:196]: #563

File: src/moscot/base/problems/compound_problem.py (hunk 14 above)
Comment (ArinaDanilina) [reply src/moscot/base/problems/compound_problem.py:296]: #565

**Comment by MUCDK** [comment]
@michalk8 I unjitted the `_apply` function for now, see https://github.com/theislab/moscot/pull/515/commits/42601e2ee855dd74a7b90270c94fbcb26944bef7. 
 How would you like to proceed with this?

File: src/moscot/base/output.py (hunk 11 above)
Comment (MUCDK) [reply src/moscot/base/output.py:144]: We decided to always use the module

**Pull request merged**